src/Definition/Oracle.php

```diff
@@ -0,0 +1,231 @@
+<?php
+namespace Keboola\Datatype\Definition;
+
+use Keboola\Datatype\Definition\Exception\InvalidLengthException;
+use Keboola\Datatype\Definition\Exception\InvalidOptionException;
+use Keboola\Datatype\Definition\Exception\InvalidTypeException;
+
+class Oracle extends Common
+{
+    const TYPES = [
+        // list of data types - https://docs.oracle.com/cd/A58617_01/server.804/a58241/ch5.htm
+        'CHAR', 'NCHAR',
+        'VARCHAR', 'NVARCHAR',
+        'VARCHAR2', 'NVARCHAR2',
+        'CLOB', 'NCLOB',
+        'LONG',
+        'NUMBER',
+        'DATE',
+        'BLOB', 'BFILE', 'RAW', 'LONG RAW'
+    ];
+
+    /**
+     * Oracle constructor.
+     * @param string $type
+     * @param array $options -- length, nullable, default
+     * @throws InvalidLengthException
+     * @throws InvalidOptionException
+     * @throws InvalidTypeException
+     */
+    public function __construct($type, $options = [])
+    {
+        $this->validateType($type);
+        $this->validateLength($type, isset($options["length"]) ? $options["length"] : null);
+        $diff = array_diff(array_keys($options), ["length", "nullable", "default"]);
+        if (count($diff) > 0) {
+            throw new InvalidOptionException("Option '{$diff[0]}' not supported");
+        }
+        parent::__construct($type, $options);
+    }
+
+    /**
+     * @return string
+     */
+    public function getSQLDefinition()
+    {
+        $definition =  $this->getType();
+        if ($this->getLength() && $this->getLength() !== "") {
+            $definition .= "(" . $this->getLength() . ")";
+        }
+        if (!$this->isNullable()) {
+            $definition .= " NOT NULL";
+        }
+        return $definition;
+    }
```

Comment:
I don't think this is required for now, but maybe good to have as maybe the writer can use it in the future.